src/index.js

```diff
@@ -43,13 +43,28 @@ let rosNode = null;
 let pingMasterTimeout = null;
 
 //------------------------------------------------------------------
-
-function _checkMasterHelper(timeout=100) {
-  const localHelper = (resolve) => {
+/**
+ * @private
+ * Helper function to see if the master is available and able to accept
+ * connections.
+ * @param {number} timeout time in ms between connection attempts
+ * @param {number} maxTimeout maximum time in seconds to retry before timing out. 
+ * A negative number will make it retry forever. 0 will only make one attempt
+ * before timing out.
+ */
+function _checkMasterHelper(timeout=100, maxTimeout=-1) {
+  let startTime = Time.now().secs;
+  const localHelper = (resolve,reject) => {
     pingMasterTimeout = setTimeout(() => {
-      // also check that the slave api server is set up
+      // also check that the slave api server is set up   
       if (!rosNode.slaveApiSetupComplete()) {
-        localHelper(resolve);
+        console.error("hello slave api isnt setup yet");
+        if (Time.now().secs - startTime >= maxTimeout && !(maxTimeout < 0) ) {
+          let error = `Unable to register with master node [${rosNode.getRosMasterUri()}]: unable to set up slave API Server. Stopping...`
```

Comment:
did you mean to do something with this variable? or am I missing where it's being used